semgrep-core/src/engine/Dataflow_tainting.ml

```diff
@@ -51,19 +51,24 @@ end)
 (*****************************************************************************)
 
 type var = Dataflow_core.var
-type deep_match = PM of Pattern_match.t | Call of G.expr * deep_match
-type source = deep_match
-type sink = deep_match
-type arg_pos = int
+type trace = G.tok list [@@deriving show]
+
+type deep_match = PM of Pattern_match.t | Call of G.expr * trace * deep_match
+[@@deriving show]
+
+type source = deep_match [@@deriving show]
+type sink = deep_match [@@deriving show]
+type arg_pos = int [@@deriving show]
 
 type finding =
-  | SrcToSink of source * sink * Metavariable.bindings
-  | SrcToReturn of source
-  | ArgToSink of arg_pos * sink
-  | ArgToReturn of arg_pos
+  | SrcToSink of source * trace * sink * Metavariable.bindings
```

Comment:
Some of these are pretty big tuples, and they're all different types; might be worth making them records

Reply:
Yeah, maybe the `finding` type needs to be re-designed, but in the meantime I added a record just for the `SrcToSink` constructor. I think it helps more in that case because `source` and `sink` are just type aliases, so there is a real possibility of mixing them up.